Sources/StreamChatUI/ImageGallery/ImageGalleryVC.swift

```diff
@@ -16,26 +16,56 @@ open class _ImageGalleryVC<ExtraData: ExtraDataTypes>:
     AppearanceProvider,
     UICollectionViewDataSource,
     UICollectionViewDelegate,
-    UICollectionViewDelegateFlowLayout {
-    /// Content to display.
-    open var content: _ChatMessage<ExtraData>! {
-        didSet {
-            updateContentIfNeeded()
+    UICollectionViewDelegateFlowLayout,
+    ComponentsProvider {
+    public enum Item {
+        case media(ChatMessageVideoAttachment)
+        case image(ChatMessageImageAttachment)
```

Comment:
Maybe an use case for the struct pattern? So that the customer can expend it?

Reply:
Very good comment! Please have a look at the latest commit, there's a WIP for handling this